### DISCUSSION TITLE
I am trying to build the app locally but i cant find documentation

### DISCUSSION THREAD

**Discussion by ymolists** [Budibase/budibase#4634] [Help] [answered]
Hi All.

Thank you so much for making this great application. I am trying to see if i can build the code locally. Is that documented ?

Thank you.

**Comment by mjashanks** [comment, marked as answer]
Thanks! Check this out: https://github.com/Budibase/budibase/blob/HEAD/.github/CONTRIBUTING.md

**Comment by ymolists** [comment]
@mjashanks Thank you so much for your prompt reply!

I was able to get the app running but i faced few issues.

* The [Contributing guide](https://github.com/Budibase/budibase/blob/HEAD/.github/CONTRIBUTING.md) does not mention post install steps. As in what to do after all the steps are completed. Maybe adding a link to https://docs.budibase.com/self-hosting/self-host which has nice post install steps would be enough ?
* When i ran `yarn setup` it was expecting that the redis server was up already which is not mentioned in the above guide. As such it was not able to connect to redis and kept failing. Did i miss a step ? i had to do `docker compose -f ./docker-compose.dev.yaml --env-file .env up` to get it to stop complaining about redis not being present
* When i went to  http://localhost:10000/builder its not askning for an admin login user. it assumes i already logged in so i am not sure what i did wrong.
* oh and yeah i am using `docker compose` (which is the recommended compose and not `docker-compose` hope thats ok
*

**Reply by mjashanks** [reply]
Thanks for reporting back your experience! Are you already running Budibase locally - i.e. via the standard self-host setup?

You need to stop all of that stack before running the dev stack. Maybe this is the case for some of your issues.

**Reply by ymolists** [reply]
Thank you for asking. I found what was the issue after all !

The problem was that i clicked  `http://localhost:3000/builder` (instead of 10000) and that i think initialised something in the db so the admin login was never possible after that. I did `yarn nuke:docker` then `docker volume prune` to prune all volumes and start from scratch and that fixed my problem.

After that I was able to login but i cant deploy the app or go into the design page its empty. Here is the error i get on the client side
```
client:74369 Uncaught ReferenceError: dayjs is not defined
    at VM915 client:74369:70
(anonymous) @ client:74369
```

```
my current version:
$git rev-parse HEAD
5862fc23222ea13a4782ecf600e95d4018eb4e22
```

I don't see any error on the server side. Maybe logging is not enabled by default ?

**Reply by ymolists** [reply]
Ok i did a pull for the latest version and restarted everything now i am able to login and design pages ... woohooo !
